File: container_with_most_water.py
from typing import List
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxArea = 0
        pointer1 = 0
        pointer2 = len(height)-1
        while(pointer1 < pointer2):
            area = min(height[pointer1], height[pointer2]) * (pointer2 - pointer1)
            if area > maxArea:
                maxArea = area
            if height[pointer1] < height[pointer2]:
                pointer1+=1
            else:
                pointer2-=1    

        return maxArea


solution = Solution().maxArea([8,7,2,1])
print(solution)     

File: containerWithMostWater.go
package main

import "fmt"

func main() {
	i := maxArea([]int{1, 8, 6, 2, 5, 4, 8, 3, 7})
	fmt.Print(i)
}

func maxArea(height []int) int {
	maximum := 0
	i, j := 0, len(height)-1

	for i < j {
		maximum = max(maximum, (j-i)*min(height[i], height[j]))
		if height[j] < height[i] {
			j--
		} else {
			i++
		}
	}
	return maximum

}
